$MirOS: ports/emulators/qemu/snapshot/patches/patch-vl_c,v 1.1 2008/05/07 15:53:13 tg Exp $
$OpenBSD: patch-vl_c,v 1.17 2008/04/28 22:52:38 todd Exp $
$NetBSD: patch-dk,v 1.4 2009/10/11 16:04:13 asau Exp $
--- vl.c.orig	Wed Sep 23 19:01:08 2009
+++ vl.c	Sat Oct 17 17:01:52 2009
@@ -567,7 +567,7 @@ static void init_get_clock(void)
 {
     use_rt_clock = 0;
 #if defined(__linux__) || (defined(__FreeBSD__) && __FreeBSD_version >= 500000) \
-    || defined(__DragonFly__)
+    || defined(__DragonFly__) || defined(__NetBSD__) || defined(__OpenBSD__)
     {
         struct timespec ts;
         if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0) {
@@ -580,7 +580,7 @@ static void init_get_clock(void)
 static int64_t get_clock(void)
 {
 #if defined(__linux__) || (defined(__FreeBSD__) && __FreeBSD_version >= 500000) \
-	|| defined(__DragonFly__)
+	|| defined(__DragonFly__) || defined(__NetBSD__) || defined(__OpenBSD__)
     if (use_rt_clock) {
         struct timespec ts;
         clock_gettime(CLOCK_MONOTONIC, &ts);
@@ -5555,8 +5555,10 @@ int main(int argc, char **argv, char **e
                     if (!strcmp(optarg, "now")) {
                         rtc_date_offset = -1;
                     } else {
+			int notm_year;
+
                         if (sscanf(optarg, "%d-%d-%dT%d:%d:%d",
-                               &tm.tm_year,
+                               &notm_year,
                                &tm.tm_mon,
                                &tm.tm_mday,
                                &tm.tm_hour,
@@ -5564,7 +5566,7 @@ int main(int argc, char **argv, char **e
                                &tm.tm_sec) == 6) {
                             /* OK */
                         } else if (sscanf(optarg, "%d-%d-%d",
-                                          &tm.tm_year,
+                                          &notm_year,
                                           &tm.tm_mon,
                                           &tm.tm_mday) == 3) {
                             tm.tm_hour = 0;
@@ -5573,7 +5575,7 @@ int main(int argc, char **argv, char **e
                         } else {
                             goto date_fail;
                         }
-                        tm.tm_year -= 1900;
+                        tm.tm_year = notm_year - 1900;
                         tm.tm_mon--;
                         rtc_start_date = mktimegm(&tm);
                         if (rtc_start_date == -1) {
